packages/gatsby/src/redux/types.ts

```diff
@@ -1,49 +1,198 @@
+import { IProgram } from "../commands/types"
 import { GraphQLSchema } from "graphql"
+import { SchemaComposer } from "graphql-compose"
+
+type SystemPath = string
+type Identifier = string
+type StructuredLog = any // TODO this should come from structured log interface
 
 export enum ProgramStatus {
   BOOTSTRAP_FINISHED = `BOOTSTRAP_FINISHED`,
   BOOTSTRAP_QUERY_RUNNING_FINISHED = `BOOTSTRAP_QUERY_RUNNING_FINISHED`,
 }
 
-export interface IReduxNode {
-  id: string
+export interface IGatsbyPage {
+  internalComponentName: string
+  path: string
+  matchPath: undefined | string
+  component: SystemPath
+  componentChunkName: string
+  isCreatedByStatefulCreatePages: boolean
+  context: {}
+  updatedAt: number
+  pluginCreator__NODE: Identifier
+  pluginCreatorId: Identifier
+  componentPath: SystemPath
+}
+
+export interface IGatsbyConfig {
+  plugins?: {
+    // This is the name of the plugin like `gatsby-plugin-manifest
+    resolve: string
+    options: {
+      [key: string]: unknown
+    }
+  }[]
+  siteMetadata?: {
+    title?: string
+    author?: string
+    description?: string
+  }
+  // @deprecated
+  polyfill?: boolean
+  developMiddleware?: any
+  proxy?: any
+  pathPrefix?: string
+}
+
+export interface INode {
```

Comment:
Why `INode` and not `IGatsbyNode` or `IPageNode`? Something a liiiitle less generic would be nice here. It was `IReduxNode`, which is a misnomer but was easy to recognize and grep for.